core/server/users.js

```diff
@@ -9,7 +9,8 @@ const mainFields = {
 }
 
 const getUsers = (levelId) => {
-    let filters = { 'status.online': true, 'profile.levelId': levelId }
+    const registeredOrGuest = {"$or": [{"profile.guest": {"$ne": true}, "username": null}, {"profile.guest": true, "username": {$exists: true}} ]}
+    const filters = { 'status.online': true, 'profile.levelId': levelId, ...registeredOrGuest}
```

Comment:
Why not just something like this?

```suggestion
    const filters = { 'status.online': true, 'profile.levelId': levelId, 'profile.guest': false }
```

Reply:
I didn't want to break the guest feature

Reply:
Also, what is `username`? In production, we have only one user with this field:

```
$ db.users.countDocuments({"username": {$exists: true}})
1
```

Looks like it's a deprecated concept we want to remove from the codebase, isn't it?

Reply:
I don't know if it is used for another purpose but it is used at least to differentiate a guest and a user before on-boarding. `username` is a random name assigned to a guest and then removed when the guest sign-in/up.

Reply:
> I didn't want to break the guest feature

Hm, fair enough, let's keep it that way for now and remove it later if we want to.